### PR TITLE
move mount / portmapping into config types

### PR DESCRIPTION
**Pull request by BenTheElder**
part of #734 
this will also make the documentation easier, as all the config types will be in the config ...

this is a quick port, we'll want to go back over this and cleanup the comments possibly

we also probably want to consider only using yaml and consider just using the on-disk format (IE strings for the enums) instead of the integers we currently use.

those can be handled at a later date though.

### CODE REVIEW

File: pkg/apis/config/v1alpha3/json.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
+/*
```
Comment (BenTheElder) [inline pkg/apis/config/v1alpha3/json.go]: this is a copy of pkg/container/cri/json.go

File: pkg/apis/config/v1alpha3/yaml.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
+/*
```
Comment (BenTheElder) [inline pkg/apis/config/v1alpha3/yaml.go]: this is a copy of pkg/container/cri/yaml.go

File: pkg/internal/apis/config/types.go [hunk 3]
```diff
@@ -130,3 +126,110 @@ type PatchJSON6902 struct {
 	// Patch should contain the contents of the json patch as a string
 	Patch string
 }
+
+// Mount specifies a host volume to mount into a container.
```
Comment (BenTheElder) [inline pkg/internal/apis/config/types.go]: these are copied from pkg/container/cri/types.go, but the struct tags have been removed (for the internal API version)

File: pkg/apis/config/v1alpha3/types.go [hunk 4]
```diff
@@ -140,3 +136,115 @@ type PatchJSON6902 struct {
 	// Patch should contain the contents of the json patch as a string
 	Patch string `yaml:"patch" json:"patch"`
 }
+
+/*
```
Comment (BenTheElder) [inline pkg/apis/config/v1alpha3/types.go]: these are copied from pkg/container/cri/types.go

**Comment by BenTheElder** [comment]
I've tested this locally. It also contains no functional changes other than the trivial conversion, the rest is just moving / copying types.

**Comment by neolit123** [comment]
LGTM, should `pkg/container/cri` be removed at this point?

**Comment by BenTheElder** [comment]
that will be next, the problem is we still use this and the docker package directly in build/, that PR may  be a bit bigger

**Comment by BenTheElder** [comment]
er pkg/build/node uses pkg/container/docker and pkg/container/cri.Mount

I'm preparing a bigger PR to clean that up and make providers viable.

**Pull request merged**